modules/ml/src/main/java/org/apache/ignite/ml/selection/scoring/metric/BinaryClassificationMetricValues.java

```diff
@@ -0,0 +1,221 @@
+
+    /** Balanced accuracy. */
+    private double balancedAccuracy;
+
+    /** F1-Score is the harmonic mean of Precision and Sensitivity. */
+    private double f1Score;
+
+    /** */
+    public double tp() {
+        return tp;
+    }
+
+    /** */
+    public void setTp(double tp) {
+        this.tp = tp;
+    }
+
+    /** */
+    public double tn() {
+        return tn;
+    }
+
+    /** */
+    public void setTn(double tn) {
+        this.tn = tn;
+    }
+
+    /** */
+    public double fp() {
+        return fp;
+    }
+
+    /** */
+    public void setFp(double fp) {
+        this.fp = fp;
+    }
+
+    /** */
+    public double fn() {
+        return fn;
+    }
+
+    /** */
+    public void setFn(double fn) {
+        this.fn = fn;
+    }
+
+    /** */
+    public double recall() {
+        return recall;
+    }
+
+    /** */
+    public void setRecall(double recall) {
+        this.recall = recall;
+    }
+
+    /** */
+    public double specificity() {
+        return specificity;
+    }
+
+    /** */
+    public void setSpecificity(double specificity) {
+        this.specificity = specificity;
+    }
+
+    /** */
+    public double precision() {
+        return precision;
+    }
+
+    /** */
+    public void setPrecision(double precision) {
+        this.precision = precision;
+    }
+
+    /** */
+    public double npv() {
+        return npv;
+    }
+
+    /** */
+    public void setNpv(double npv) {
+        this.npv = npv;
+    }
+
+    /** */
+    public double fallOut() {
+        return fallOut;
+    }
+
+    /** */
+    public void setFallOut(double fallOut) {
+        this.fallOut = fallOut;
+    }
+
+    /** */
+    public double fdr() {
+        return fdr;
+    }
+
+    /** */
+    public void setFdr(double fdr) {
+        this.fdr = fdr;
+    }
+
+    /** */
+    public double missRate() {
+        return missRate;
+    }
+
+    /** */
+    public void setMissRate(double missRate) {
+        this.missRate = missRate;
+    }
+
+    /** */
+    public double accuracy() {
+        return accuracy;
+    }
+
+    /** */
+    public void setAccuracy(double accuracy) {
+        this.accuracy = accuracy;
+    }
+
+    /** */
+    public double balancedAccuracy() {
+        return balancedAccuracy;
+    }
+
+    /** */
+    public void setBalancedAccuracy(double balancedAccuracy) {
+        this.balancedAccuracy = balancedAccuracy;
+    }
+
+    /** */
+    public double f1Score() {
+        return f1Score;
+    }
+
+    /** */
+    public void setF1Score(double f1Score) {
+        this.f1Score = f1Score;
+    }
+
+    /** Returns the pair of metric name and metric value. */
+    public Map<String, Double> toMap() {
+        Map<String, Double> metricValues = new HashMap<>();
+        for (Field field : this.getClass().getDeclaredFields())
+            try {
+                metricValues.put(field.getName(), field.getDouble(this));
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
```

Comment:
I don't think that exception hiding is valid behavior for this class

Reply:
Should we rethrow something IgniteML related do yu think?